### ISSUE TITLE
Settle minimum browser version required

### ISSUE THREAD

**Issue by Kebap** [Mudlet/mudlet-package-repo#98]
Any ideas where we draw lines?

Some folks like to play Mudlet on old Win XP or Linux versions from years ago. These will probably ship outdated browsers as well.

How far back do we need ir want to support them?

This may influence ability to use certain features or no.

**Comment by vadi2** [comment]
I'd say whatever win7 supports - so ie11. Respective Firefox and Chrome versions too.

**Comment by SlySven** [comment]
How do you set a limit - or is it a question of not using things that older browsers won't handle? Also does it involve testing something in the End-users browser?  I hope it doesn't come down to agent-string sniffing as that is just not reliable...

**Comment by vadi2** [comment]
Right, just a matter of how much of a headache are we willing to do to support older browsers.

**Comment by Kebap** [comment]
> not using things that older browsers won't handle

This.
Which bleeding edge features can we use?
At what point do we stop paying attention to older browsers?
When will we say "no" to a person asking for help playing Mudlet on their grandma's old hardware?
Obviously the Mudlet client as an application can be tweaked, but this repo relies on the browser.

**Comment by vadi2** [comment]
There's a great website https://caniuse.com for this, once we settle on the minimum browser version.

**Comment by vadi2** [comment]
Since the start of the year, these are the browsers used on our websites that have passed the 1% barrier:

![image](https://user-images.githubusercontent.com/110988/83937842-88e58200-a7d0-11ea-9654-21a92d8a5213.png)

I think the worst browser we need to worry about is Internet Explorer. Unfortunately the stats can't show us what version of it are people using, but I think whatever version an updated Windows 7 installation has would be good enough.

**Comment by Kebap** [comment]
I think Matomo can list browser versions as well... need to remember where exactly :)

**Comment by Kebap** [comment]
> whatever win7 supports - so ie11. Respective Firefox and Chrome versions too.

* Internet Explorer - [v11](https://www.microsoft.com/en-us/download/details.aspx?id=41628)
  * Even though this version is not updated by Mircosoft anymore (they since switched to Edge) Win7 is still running this
* Firefox - [current](https://www.mozilla.org/en-US/firefox/system-requirements/)
* Chrome - [current](https://www.google.com/chrome/)
  * Google [has promised](https://cloud.google.com/blog/products/chrome-enterprise/how-chrome-is-helping-enterprises-still-using-windows-7) to keep new Chrome versions working on Win 7 until at least 2021-07

**Comment by vadi2** [comment]
That sounds good to me!

**Comment by SlySven** [comment]
For FireFox you'd probably also want to handle the ESR version.

*As an aside GitHub is playing silly buggers in that they have go so bleeding edge that they are not supporting my browser choice any-more - though someone has come up with an XPI (not something that Mozilla touches nowadays) addon https://github.com/JustOff/github-wc-polyfill/ that fixes things (since the person released it a few days ago they have broken things further and an update to the fix has become necessary) - what is scary is that GitHub is the location of the repository for the browser so that they (GH) are helping to prevent the developer not to be able eat their own dog food...*
